### PR TITLE
[Execution] New ingestion engine

### PR DESCRIPTION
**Pull request by zhangchiqing**
Please review the following PRs first:

https://github.com/onflow/flow-go/pull/5248
https://github.com/onflow/flow-go/pull/5288
https://github.com/onflow/flow-go/pull/5337

This PR adds a new ingestion engine (Ingestion.NewMachine) that is responsible for fetching data for unexecuted blocks.

The new ingestion engine could resolve the following issues:
https://github.com/onflow/flow-go/issues/5161
https://github.com/onflow/flow-go/issues/5217
https://github.com/onflow/flow-go/issues/5298

The new ingestion engine can be turned on with a new flag `--enable-new-ingestion-engine=true`.

The new flag allows us to switch back to the old ingestion engine during the testing phase if the new engine ran into any problem, such as the block execution is halted. 

It would not cause execution fork, because the ingestion engine is only responsible for determine when a block can be executed, rather than how a block is executed.

### CODE REVIEW

File: cmd/Dockerfile [hunk 1]
```diff
@@ -37,7 +37,7 @@ WORKDIR /app
 
 ARG GOARCH=amd64
 # TAGS can be overriden to modify the go build tags (e.g. build without netgo)
-ARG TAGS="netgo"
+ARG TAGS="netgo,osusergo"
```
Comment (peterargue) [inline cmd/Dockerfile:40]: I'm ok with updating the default user, but I think this should go in a different PR since it has implications for node operators and infra

File: engine/execution/ingestion/core.go [hunk 2]
```diff
@@ -164,6 +164,8 @@ func (e *Core) launchWorkerToExecuteBlocks(ctx irrecoverable.SignalerContext, re
 }
 
 func (e *Core) OnBlock(header *flow.Header, qc *flow.QuorumCertificate) {
+	e.log.Debug().Msgf("received block %v (%v)", header.Height, qc.BlockID)
```
Comment (peterargue) [inline engine/execution/ingestion/core.go:167]: make the args fields?

**Review by peterargue — APPROVED** [review]

File: cmd/execution_config.go [hunk 3]
```diff
@@ -120,6 +121,7 @@ func (exeConf *ExecutionConfig) SetupFlags(flags *pflag.FlagSet) {
 	flags.BoolVar(&exeConf.onflowOnlyLNs, "temp-onflow-only-lns", false, "do not use unless required. forces node to only request collections from onflow collection nodes")
 	flags.BoolVar(&exeConf.enableStorehouse, "enable-storehouse", false, "enable storehouse to store registers on disk, default is false")
 	flags.BoolVar(&exeConf.enableChecker, "enable-checker", true, "enable checker to check the correctness of the execution result, default is true")
+	flags.BoolVar(&exeConf.enableNewIngestionEngine, "enable-new-ingestion-engine", false, "enable new ingestion engine, default is false")
```
Comment (zhangchiqing) [inline cmd/execution_config.go:124]: By default the new ingestion engine is disabled, and can be enabled by flag.

File: integration/localnet/builder/bootstrap.go [hunk 4]
```diff
@@ -399,11 +399,14 @@ func prepareExecutionService(container testnet.ContainerConfig, i int, n int) Se
 		panic(err)
 	}
 
+	enableNewIngestionEngine := true
```
Comment (zhangchiqing) [inline integration/localnet/builder/bootstrap.go:402]: the localnet enables new ingestion engine by default

**Review by janezpodhostnik — APPROVED** [review]
